teps/0044-composing-tasks-with-tasks.md

```diff
@@ -0,0 +1,137 @@
+  it can't use a different kind of PipelineResource to provide those files (tho you can avoid
+  doing the clone multiple times using [from](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-the-from-parameter))
+
+### Goals
+
+- Make it possible to combine Tasks together so that you can run multiple
+  Tasks as "one unit" (see [Requirements](#requirements):
+  - At runtime in a TaskRun
+  - At Pipeline authoring time
+- Add some of [the features we don't have without PipelineResources](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#)
+  to Tekton Pipelines (without requiring use of PipelineResources)
+  - If possible we can provide a migration path from PipelineResources to the solution proposed here, where applicable
+
+### Non-Goals
+
+- Composing Tasks within Tasks at [Task authoring time](https://github.com/tektoncd/community/blob/master/design-principles.md#reusability).
+  We could decide to include this if we have some use cases that need it; for now avoiding
+  this allows us to avoid many layers of nesting (i.e. Task1 uses Task2 uses Task3, etc.)
+  or even worse, recursion (Task 1 uses Task 2 uses Task 1...)
+- Completely replacing PipelineResources: we could decide to solve this by improving PipelineResources,
+  or we could add a new feature via this TEP and still continue to support PipelineResources
+  (since they provide [more functionality than just composition](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#))
+
+### Use Cases (optional)
+
+- A user wants to use catalog Tasks to checkout code, run unit tests and upload results,
+  and does not want to incur the additional overhead (and performance impact) of creating
+  volume based workspaces to share data between them in a Pipeline. e.g. specifically
+  cloning with [git-clone](https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.2),
+  running tests with [golang-test](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1)
+  and uploading results with [gcs-upload](https://github.com/tektoncd/catalog/tree/master/task/gcs-upload/0.1).
+- A user wants to use a Task from the catalog with a git repo and doesn't want the extra
+  overhead of using a Pipeline, they just want to make a TaskRun,
+  e.g. [@mattmoor's feedback on PipelineResources and the Pipeline beta](https://twitter.com/mattomata/status/1251378751515922432))
+  where he wants to checkout some code and [use the kaniko task](https://github.com/tektoncd/catalog/tree/master/task/kaniko/0.1)
+  without having to fiddle with volumes
+- An organziation does not want to use PVCs at all; for example perhaps they have decided
+  on uploading to and downloading from buckets in the cloud (e.g. GCS)
+- An organization is willing to use PVCs to some extent but needs to put limits on their use
+
+## Requirements
+
+- Tasks can be composed together run as "one unit":
```

Comment:
In https://github.com/tektoncd/community/pull/318 this "grouping/composition" is done by the fact that the Tasks **share workspace** - so no new things for the Pipeline author - perhaps other than using an emtpyDir-workspace instead of a PVC-workspace in the PipelineRun.

Other than this, the _Requirements_ of these two TEPs should be aligned now.